addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -698,32 +698,47 @@ export class OdooEditor extends EventTarget {
                 this.updateColorpickerLabels();
             });
         }
-        // Handle the font size input.
         const fontSizeInput = this.toolbar.querySelector('input#fontSizeCurrentValue');
         this.addDomListener(this.toolbar, 'click', ev => {
             if (fontSizeInput && ev.target.closest('#font-size .dropdown-toggle')) {
-                this._lastMousedown = undefined;
                 // If the click opened the font size dropdown, select the input content.
                 fontSizeInput.select();
             } else if (!this.isSelectionInEditable()) {
                 // Otherwise, if we lost the selection in the editable, restore it.
                 this.historyResetLatestComputedSelection(true);
             }
         })
+        // Handle the font size input.
         if (fontSizeInput) {
-            this.addDomListener(this.document, 'mousedown', ev => (this._lastMousedown = ev.target));
-            this.addDomListener(fontSizeInput.ownerDocument, 'mousedown', ev => (this._lastMousedown = ev.target));
-            this.addDomListener(fontSizeInput, 'change', () => {
-                // Don't apply if the user clicked on one of the dropdown items.
-                if (!this._lastMousedown?.closest('#font-size .dropdown-item') && parseInt(fontSizeInput.value) > 0) {
+            let hasInputChanged = false;
+            const onInputChange = () => {
+                if (fontSizeInput.value === 'default' || parseInt(fontSizeInput.value) > 0) {
                     if (!this.isSelectionInEditable()) {
                         this.historyResetLatestComputedSelection(true);
                     }
-                    this.execCommand('setFontSize', parseInt(fontSizeInput.value) + 'px');
+                    const value = fontSizeInput.value === 'default' ? undefined : parseInt(fontSizeInput.value) + 'px';
+                    this.execCommand('setFontSize', value);
                     fontSizeInput.closest('#font-size').querySelectorAll('.show').forEach(el => el.classList.remove('show'));
                     fontSizeInput.blur();
+                    hasInputChanged = false;
                 }
-            });
+            };
+            this.addDomListener(fontSizeInput, 'change', onInputChange);
+            this.addDomListener(fontSizeInput, 'input', () => { hasInputChanged = true; });
+            // Handle the font size dropdown.
+            const fontSizeDropdown = this.toolbar.querySelector('#font-size');
+            if (fontSizeDropdown) {
+                fontSizeDropdown.querySelectorAll('.dropdown-item').forEach(item => {
+                    this.addDomListener(item, 'mousedown', ev => {
+                        fontSizeInput.value = ev.target.textContent;
+                        if (!hasInputChanged) {
+                            // If the input value has not been changed, call the
+                            // change event listener manually.
```

Comment:
tldr; everything goes to hell when we call `hasInputChanged` twice, for weird browser reasons.


If the input changed (I typed something in the input) and I click on a dropdown item, that click will end up triggering a change event so `onInputChange` will be called. So if I call it manually in that case, it will be executed twice.
`onInputChange` will call `historyResetLatestComputedSelection`, which will end up calling `setSelection`, which does `selection.setRange` before calling `extend` on that selection. But `setRange` triggers a `selectionchange` event and we set the latest computed selection before having finished setting the selection (so it's still collapsed), and finally we enter `onInputChange` a second time, call `historyResetLatestComputedSelection` again, which sets a wrong selection. It works anyway, but the toolbar disappears because the selection is now collapsed.

Reply:
:facepalm: That should work yes ^^'

Reply:
Done.